.Lib9c.Tests/Model/ItemNotificationTest.cs

```diff
@@ -143,9 +143,44 @@ public void EquipTwoRings()
             var strongests = ordered.TakeLast(2).Select(i => i.ItemId);
             equipmentList.AddRange(strongests);
             avatarState.EquipEquipments(equipmentList);
-            hasNotification = inventory.HasNotification(avatarState.level);
+            hasNotification = inventory.HasNotification(avatarState.level, 0);
             // When the 1st strongest, the 2nd strongest items are equipped.
             Assert.False(hasNotification);
         }
+
+        [Fact]
+        public void RequiredBlockIndexTest()
+        {
+            var avatarAddress = _agentAddress.Derive("avatar_2");
+            var avatarState = new AvatarState(
+                avatarAddress,
+                _agentAddress,
+                0,
+                _tableSheets.GetAvatarSheets(),
+                new GameConfigState(),
+                default
+            );
+
+            avatarState.level = GameConfig.RequireCharacterLevel.CharacterEquipmentSlotRing2;
+            var equipmentList = new List<Guid>();
+            avatarState.EquipEquipments(equipmentList);
+
+            var inventory = avatarState.inventory;
+            var hasNotification = inventory.HasNotification(avatarState.level, 0);
+            // When inventory is empty.
+            Assert.False(hasNotification);
+
+            var rows = _tableSheets.EquipmentItemSheet.Values
+                .Where(r => r.ItemSubType == ItemSubType.Ring && r.Grade == 1);
+            foreach (var row in rows)
+            {
+                var guid = Guid.NewGuid();
+                var equipment = (Equipment)ItemFactory.CreateItemUsable(row, guid, 1, 0);
+                inventory.AddItem(equipment);
+            }
+
+            hasNotification = inventory.HasNotification(avatarState.level, 0);
+            Assert.False(hasNotification);
```

Comment:
Shouldn't it be'True' after a ring is added to your inventory?

Reply:
`var equipment = (Equipment)ItemFactory.CreateItemUsable(row, guid, 1, 0);`
As you can see, its `RequiredIndex` is 1, so it should be false.

Reply:
Now I see.
How about making it easier to read using the `Theory` and `InlineData` attributes?
ex.
```
[Theory]
[InlineData(0, 0, true)]
[InlineData(0, 1, false)]
[InlineData(...)]
public void RequiredBlockIndexTest(long blockIndex, long requiredBlockIndex, bool hasNotification)
{ /* */ }
```

Reply:
Seems like a good idea!